.github/workflows/auto-pr-submodules.yml

```diff
@@ -0,0 +1,38 @@
+# This is a basic workflow to help you get started with Actions
+name: Update Submodules and Submit a PR
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push  only for the actions-test-branch branches
+  push:
+    branches:
+      - master
+    # Runs every day at midnight UTC
+    schedule:
+      - cron:  '0 0 * * *'
+
+jobs:
+  createPullRequest:
+    if: ${{ contains(github.ref, 'master') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: submodules-init
+        uses: snickerbockers/submodules-init@v4
+
+      - name: Update All Submodules to Remote Latest
+        run: git submodule update --remote
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: Update All Submodules to remote latest
```

Comment:
I would make the commit message and the title match. Maybe these params can be created in a previous step and then used here. That's would allow to customize the message/body dependencing on which submodules were updated.

Moreover, `hub` is preinstalled in the virtual environments. Might be worth having a look at it, instead of dependending on a third-party Action.

Reply:
Can we make the Commit message look similar to the ones in this pull request --> https://github.com/google/skywater-pdk/pull/233

Reply:
I *think* this was the Python code which generated that status message,
```python
        log = subprocess.check_output(['git', 'log', '--graph', '--pretty=short', '--decorate=short', '--decorate-refs=remotes/origin', '--shortstat', old_version+'..'+new_version], cwd=submod).decode('utf-8')
```

```python
output = ['Updating submodules on {} UTC'.format(now), '']

for lib, ver, ov, nv in changes:
    if ov == nv:
        continue
    output.append('''\
 - Updating [`{lib}` {ver}](https://foss-eda-tools.googlesource.com/skywater-pdk/libs/{lib}/+/{ov}..{nv}) to {nv}'''.format(
     lib=lib, ver=ver, ov=ov, nv=nv))

output.append('')
output.append(get_submod_info())

print('\n'.join(output))
```

Reply:
In the workflow, the required change would be:

```yml

Matrix:
  runs-on: ubuntu-latest
  outputs:
    matrix: ${{ steps.generate.outputs.matrix }}
  steps:
  - uses: actions/checkout@v2
  - id: generate
    run: SCRIPT_FOR_GENERATING_THE_ARGS
    
...
    
- uses: peter-evans/create-pull-request@v3
  with: ${{ fromJson(needs.Matrix.outputs.matrix) }}
```

Reply:
@mithro, @umarcor: I changed the script that @mithro gave me a bit, and now it handles the commit message, then in the next step I'm exporting the PR body to be the same as the commit message. Can you take a look?